crates/oxc_linter/src/rules/unicorn/catch_error_name.rs

```diff
@@ -0,0 +1,272 @@
+use oxc_ast::{
+    ast::{Argument, BindingPatternKind, Expression},
+    AstKind,
+};
+use oxc_diagnostics::{
+    miette::{self, Diagnostic},
+    thiserror::Error,
+};
+use oxc_macros::declare_oxc_lint;
+use oxc_semantic::SymbolId;
+use oxc_span::{Atom, Span};
+
+use crate::{context::LintContext, rule::Rule, AstNode};
+
+#[derive(Debug, Error, Diagnostic)]
+#[error("eslint-plugin-unicorn(catch-error-name): The catch parameter {0:?} should be named {1:?}")]
+struct CatchErrorNameDiagnostic(Atom, Atom, #[label] pub Span);
+
+#[derive(Debug, Clone)]
+pub struct CatchErrorName {
+    ignore: Vec<Atom>,
+    name: Atom,
+}
+
+impl Default for CatchErrorName {
+    fn default() -> Self {
+        Self { ignore: vec![], name: Atom::new_inline("error") }
+    }
+}
+
+declare_oxc_lint!(
+    /// ### What it does
+    ///
+    /// This rule enforces naming conventions for catch statements.
+    ///
+    /// ### Example
+    /// ```javascript
+    ///
+    /// // fail
+    /// try { } catch (foo) { }
+    ///
+    /// // pass
+    /// try { } catch (error) { }
+    ///
+    /// ```
+    CatchErrorName,
+    correctness
+);
+
+impl CatchErrorName {
```

Comment:
nit: can you move this code block below the `impl Rule` block? This will make this file a little bit easier to read by having more important information at the top.

Reply:
yeah makes sense